docs/design/PicoD-Design.md

```diff
@@ -0,0 +1,674 @@
+
+The authentication model balances security with operational simplicity, using client-generated tokens and one-time initialization to bind each sandbox securely to its designated end user.
+
+##### Authentication Architecture
+
+```mermaid
+sequenceDiagram
+    participant Client as SDK Client
+    participant Frontend as AgentCube Frontend
+    participant SandboxMgr as Sandbox Manager
+    participant PicoD as PicoD Daemon
+    
+
+    Note over Client, PicoD: Sandbox Creating & Frontend Auth Initialization
+    Frontend-->>Frontend: Key Pair Generation
+    SandboxMgr-->>PicoD: Create CodeInterpreter Sandbox with frontend public key embedded
+    
+    
+    Note over Client, PicoD: Sandbox Allocation & Initialization
+    Client->>Frontend: POST /api/StartCodeInterpreter (with client public key)
+    Frontend->>PicoD: Forward POST /api/init (with client public key & auth by frontend key pair)
+    PicoD->>PicoD: Encrypt & store credentials locally
+    PicoD-->>Frontend: 200 OK (init successful)
+    Frontend-->>Client: 200 OK (init successful)
+    
+    Note over Client, PicoD: Authenticated Operations
+    Client->>PicoD: POST /api/execute (Authorization: client key pair)
+    PicoD->>PicoD: Validate token against stored credentials
+    PicoD-->>Client: Command execution result
+    
+    Client->>PicoD: POST /api/files (Authorization: client key pair)
+    PicoD->>PicoD: Validate token
+    PicoD-->>Client: File operation result
+```
+
+##### Security Considerations
+
+**1. One-Time Initialization**
+- Init interface can only be called once per sandbox lifecycle
+- Credentials cannot be modified after initial setup
+- Implementation includes atomic file operations to prevent race conditions
+
+**2. Frontend-Only Access Control**
+- Use pre-embedded key pair
+- Init endpoint restricted to internal network calls from AgentCube frontend
+- Rate limiting on init endpoint to prevent brute force attempts
+
+**3. Credential Security**
+- Client-generated tokens/keypairs ensure frontend never stores user credentials
+- Local encryption prevents credential exposure if container is compromised
+- Automatic credential cleanup on container termination
+
+**4. Warmpool Compatibility**
+- Containers start without authentication credentials
+- Init interface called only when sandbox allocated to specific user
+
+##### Core Authentication Components
+
+**1. Initialization Interface**
+
+- **Endpoint**: `POST /api/init`
+- **Purpose**: One-time setup of authentication credentials when sandbox is allocated to end user
+- **Access Control**: Frontend-only (protected against external access)
+- **Request Body**:
+
+```json
+{
+  "auth_type": "token|keypair",
+  "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9...",
```

Comment:
what is this token? If this is the default token used by picode to auth agent-gateway, we should take it in header